project/src/test/java/com/google/sps/data/PlaceTest.java

```diff
@@ -52,24 +60,35 @@ public void create_invalidLowPriceLevel_throwsIllegalArgumentException() {
   }
 
   @Test
-  public void create_invalidHighPriceLevel_throwsIllegalArgumentException() {
+  public void build_invalidHighPriceLevel_throwsIllegalArgumentException() {
     int invalidHighPriceLevel = 5;
 
     assertThrows(IllegalArgumentException.class, () -> {
       getValidPlaceBuilder().setPriceLevel(invalidHighPriceLevel).build();
     });
   }
 
-  /**
-   * @return a Place builder that has valid values of all attributes
-   */
+  @Test
+  public void build_validInput_returnsValidPlace() {
+    Place place = getValidPlaceBuilder().build();
+      assertAll("place",
```

Comment:
Why do you need this `assertAll` syntax? 
Why not:
```java
assertEquals(NAME, place.name());
assertEquals(WEBSITE, place.websiteUrl());
...
```

Reply:
I read on some online articles that it is best to use assertAll when asserting a set of properties that belong together conceptionally. The advantage is that it always checks all of the assertions that are passed to it, no matter how many fail and then you get a detailed result of all that went wrong. Without it the first assertEquals that goes wrong will stop the test.

Reply:
Interesting! In Google that pattern it's not used, but I see the value in it. 
In C++ we have two different methods for testing: `EXPECT_THAT`, which does not stop the test if it fails; and `ASSERT_THAT`, which does stop the test. Then, we use `ASSERT` only if the condition we check is a precondition to other tests. For example:
```c
ASSERT_THAT(array, SizeIs(1)); // If this will continue execution, the next statement will fail since array[1] doesn't exist.
EXPECT_THAT(array[0], HasSomething());
```
There is something similar in Java but it's not widely used: https://truth.dev/api/0.42/com/google/common/truth/Expect.html

It's basically the same pattern, but it's used mainly for slow tests.